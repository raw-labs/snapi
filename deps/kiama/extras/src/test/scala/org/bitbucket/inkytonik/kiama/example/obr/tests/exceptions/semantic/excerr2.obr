(* Exception-related errors in statements *)

PROGRAM ExcErr2 (i : INTEGER) : INTEGER;

CONST
    FallOverALot : EXCEPTION;

VAR
    j : INTEGER;

BEGIN

    (* Exceptions used in assignment *)
    FallOverALot := 1;
    DivideByZero := 10;

    (* Exception used in expressions *)
    IF FallOverALot > 10 THEN
        i := DivideByZero + 1;
    END

    (* RAISE a non-exception *)
    RAISE i;

    (* CATCH a non-exception *)
    TRY
        i := 0;
    CATCH j DO
        i := 10;
    CATCH i DO
        i := 10;
    END
    
    (* Find type errors in TRY body and CATCH clause bodies *)
    TRY
        i := true;
    CATCH DivideByZero DO
        FallOverALot := 1;
    CATCH FallOverALot DO
        i := j + DivideByZero;
    END
    
END ExcErr2.
