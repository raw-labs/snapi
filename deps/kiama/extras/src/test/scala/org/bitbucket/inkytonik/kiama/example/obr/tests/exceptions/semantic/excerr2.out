src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:14:5:error: illegal assignment
    FallOverALot := 1;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:14:21:error: expected exception type got integer
    FallOverALot := 1;
                    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:15:5:error: illegal assignment
    DivideByZero := 10;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:15:21:error: expected exception type got integer
    DivideByZero := 10;
                    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:18:8:error: expected integer or any enumeration type got exception
    IF FallOverALot > 10 THEN
       ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:19:14:error: expected integer type got exception
        i := DivideByZero + 1;
             ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:23:5:error: raise parameter i must be an exception constant
    RAISE i;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:28:5:error: catch clause parameter j must be an exception constant
    CATCH j DO
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:30:5:error: catch clause parameter i must be an exception constant
    CATCH i DO
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:36:14:error: true is not declared
        i := true;
             ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:38:9:error: illegal assignment
        FallOverALot := 1;
        ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:38:25:error: expected exception type got integer
        FallOverALot := 1;
                        ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/exceptions/semantic/excerr2.obr:40:18:error: expected integer type got exception
        i := j + DivideByZero;
                 ^
